pkg/api/addons.go

```diff
@@ -254,6 +255,10 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		},
 	}
 
+	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.16.0") {
+		defaultMetricsServerAddonsConfig.Mode = AddonModeReconcile
```

Comment:
Was there a good reason for `EnsureExists` here before 1.16? It's odd that they differed.

Also if `Reconcile` is the default from 1.16 onward, maybe it should be the default and this block could set `EnsureExists` for earlier versions. Then when 1.16 support drops off, this block could be deleted altogether.

Reply:
I can imagine that we decided to prefer `Reconcile` at a time when we were introducing 1.15, and didn't want to change pre-existing versions?

I like your idea.